frontend/src/user/profiles/UserProfile.tsx

```diff
@@ -0,0 +1,197 @@
+import React, { useEffect, useState } from 'react';
+import { useDispatch, useSelector } from 'react-redux';
+import { Link, useParams } from 'react-router-dom';
+import { useCookies } from 'react-cookie';
+import { Nav, Container, Loader, Notification } from '@components/';
+import { UserIcon } from '@icons/';
+import { RootState } from '@store/store';
+import { FollowerRecord, FollowingRecord } from '@store/user';
+import {
+  followUser,
+  getUser,
+  getUserFollowers,
+  getUserFollowing,
+  unfollowUser,
+} from '@api/UserApi';
+import UserProfileItem from './UserProfileItem';
+import { setIsError, setIsLoading } from '@store/category';
+
+interface SpecificUser {
+  id?: number;
+  name?: string;
+  user_image?: string;
+  email?: string;
+  followers?: FollowerRecord[];
+  following?: FollowingRecord[];
+}
+
+export default function UserProfile() {
+  const { user_id } = useParams();
+  const [cookies, _] = useCookies();
+  const dispatch = useDispatch();
+  const state = useSelector((state: RootState) => state);
+  const [user, setUser] = useState<SpecificUser>();
+  const [isViewingFollowers, setIsViewingFollowers] = useState(true);
+  const isFollowing = JSON.parse(
+    localStorage.getItem('user_following') || '{}'
+  ).find((following: FollowingRecord) => {
+    return following.following[0].id === parseInt(user_id ?? '');
+  });
+
+  const _getUser = (user_id: number) => {
+    getUser(user_id)
+      .then((response) => {
+        const { id, name, email, user_image } = response.data.data;
+        setUser({
+          id: id,
+          name: name,
+          email: email,
+          user_image: user_image,
+        });
+      })
+      .catch(() => dispatch(setIsError(true)));
+  };
+
+  const _getFollowersOfUser = (user_id: number) => {
+    getUserFollowers(user_id)
+      .then((response) => {
+        const user_followers = response.data.data.filter(
+          (follower: FollowerRecord) => {
+            return follower.follower[0].id !== cookies.user['id'];
+          }
+        );
+        setUser((value) => ({ ...value, followers: user_followers }));
+        dispatch(setIsLoading(false));
+      })
+      .catch();
+  };
+
+  const _getFollowingOfUser = (user_id: number) => {
+    getUserFollowing(user_id)
+      .then((response) => {
+        const user_following = response.data.data.filter(
+          (following: FollowingRecord) => {
+            return following.following[0].id !== cookies.user['id'];
+          }
+        );
+        setUser({ ...user, following: user_following });
+      })
+      .catch();
+  };
+
+  const handleFollowUser = (user_id: number) => {
+    dispatch(setIsLoading(true));
+    const request = isFollowing
+      ? unfollowUser(cookies.user['id'], user_id)
+      : followUser(cookies.user['id'], { following_id: user_id });
```

Comment:
This kind of condition sometimes may not be secure.
Please create totally different handler for following and unfollowing, and call these function directly in the rendered jsx if needed

Reply:
noted miss, will change